Framework/DataHandling/test/LoadILLPolarizedDiffractionTest.h

```diff
@@ -0,0 +1,523 @@
+
+    TS_ASSERT_DELTA(workspaceEntry1->x(130)[0], 2.84, 0.01)
+    TS_ASSERT_DELTA(workspaceEntry1->x(130)[1], 3.47, 0.01)
+    TS_ASSERT_EQUALS(workspaceEntry1->y(130)[0], 4)
+    TS_ASSERT_DELTA(workspaceEntry1->e(130)[0], 2.00, 0.01)
+
+    TS_ASSERT_DELTA(workspaceEntry1->x(131)[0], 2.84, 0.01)
+    TS_ASSERT_DELTA(workspaceEntry1->x(131)[1], 3.47, 0.01)
+    TS_ASSERT_EQUALS(workspaceEntry1->y(131)[0], 17)
+    TS_ASSERT_DELTA(workspaceEntry1->e(131)[0], 4.12, 0.01)
+
+    TS_ASSERT_DELTA(workspaceEntry1->x(132)[0], 2.84, 0.01)
+    TS_ASSERT_DELTA(workspaceEntry1->x(132)[1], 3.47, 0.01)
+    TS_ASSERT_EQUALS(workspaceEntry1->y(132)[0], 167943)
+    TS_ASSERT_DELTA(workspaceEntry1->e(132)[0], 409.80, 0.01)
+
+    TS_ASSERT_DELTA(workspaceEntry1->x(133)[0], 2.84, 0.01)
+    TS_ASSERT_DELTA(workspaceEntry1->x(133)[1], 3.47, 0.01)
+    TS_ASSERT_EQUALS(workspaceEntry1->y(133)[0], 2042)
+    TS_ASSERT_DELTA(workspaceEntry1->e(133)[0], 45.18, 0.01)
+
+    MatrixWorkspace_sptr workspaceEntry5 =
+        std::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(
+            outputWS->getItem(5));
+    TS_ASSERT(workspaceEntry5)
+    TS_ASSERT_EQUALS(workspaceEntry5->getNumberHistograms(), 134)
+    TS_ASSERT_EQUALS(workspaceEntry5->blocksize(), 1)
+    TS_ASSERT(workspaceEntry5->detectorInfo().isMonitor(132))
+    TS_ASSERT(workspaceEntry5->detectorInfo().isMonitor(133))
+    TS_ASSERT(workspaceEntry5->isHistogramData())
+    TS_ASSERT(!workspaceEntry5->isDistribution())
+  }
+
+  void test_D7_timeOfFlight() {
+    LoadILLPolarizedDiffraction alg;
+    // Don't put output in ADS by default
+    alg.setChild(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("Filename", "ILL/D7/395850.nxs"))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("OutputWorkspace", "_unused_for_child"))
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("PositionCalibration", "None"))
+    TS_ASSERT_THROWS_NOTHING(alg.execute())
+    TS_ASSERT(alg.isExecuted())
+
+    WorkspaceGroup_sptr outputWS = alg.getProperty("OutputWorkspace");
+    TS_ASSERT(outputWS)
+    TS_ASSERT_EQUALS(outputWS->getNumberOfEntries(), 2)
+    MatrixWorkspace_sptr workspaceEntry1 =
+        std::dynamic_pointer_cast<Mantid::API::MatrixWorkspace>(
+            outputWS->getItem(0));
+    TS_ASSERT(workspaceEntry1)
+    TS_ASSERT_EQUALS(workspaceEntry1->getNumberHistograms(), 134)
+    TS_ASSERT_EQUALS(workspaceEntry1->blocksize(), 512)
+    TS_ASSERT(workspaceEntry1->detectorInfo().isMonitor(132))
+    TS_ASSERT(workspaceEntry1->detectorInfo().isMonitor(133))
+    TS_ASSERT(workspaceEntry1->isHistogramData())
+    TS_ASSERT(!workspaceEntry1->isDistribution())
+
+    TS_ASSERT_EQUALS(workspaceEntry1->getAxis(0)->unit()->caption(), "Time")
+    TS_ASSERT_EQUALS(workspaceEntry1->YUnitLabel(), "Counts")
+
+    TS_ASSERT_DELTA(workspaceEntry1->x(0)[0], 180.00, 0.01)
```

Comment:
Those checks can be done on all the entries in the group in a loop. Same applies to other tests.